### PR TITLE
[Modelzoo] Add template for rebuilding.

### PR DESCRIPTION
**Pull request by Duyi-Wang**

### CODE REVIEW

File: modelzoo/template.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        self.label = None
+        self.feature = None
+        self.bf16 = False
+        self._is_training = True
+
+        self.input_fn()
+        self.model_fn()
+        self.train_fn()
+        self.metrics_fn()
+
```
Comment (pujiang2018) [inline modelzoo/template.py:26]: i don't think we need to call into the function like input_fn, train_fn, etc.

File: modelzoo/template.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    # process input data
+    def input_fn(self):
```
Comment (pujiang2018) [inline modelzoo/template.py:34]: suggest take input_fn out to make every model has same/similar input function

File: modelzoo/template.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        pass
+
+    # model structure
+    # finally get a probability of predict
```
Comment (pujiang2018) [inline modelzoo/template.py:38]: Is the comment correct? I think it aims to create a model.

File: modelzoo/template.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+
+# Get parse
+def get_arg_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument()
+    return parser
+
+
+def train(session, model, dataset, steps):
```
Comment (pujiang2018) [inline modelzoo/template.py:85]: we need to make it more flexible, like how many steps to save the checkpoint.

File: modelzoo/template.py [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+    print('Batch size, epoch, steps')
+
+    # set fixed random seed
+    print('Random seed')
+
+    # set directory path for checkpoint_dir
+    print('Checkpoint directory path')
+
+    # create feature column
+    feature_column = build_feature_cols()
```
Comment (pujiang2018) [inline modelzoo/template.py]: where is this feathre_column used? should this be included when creating the model?

File: modelzoo/template.py [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+
+    # create data pipline of train & test dataset
+    train_dataset = generate_input_data()
+    test_dataset = generate_input_data()
```
Comment (pujiang2018) [inline modelzoo/template.py:122]: can we take input_fn out to cover this?

File: modelzoo/template.py (hunk 4 above)
Comment (Duyi-Wang) [reply modelzoo/template.py:85]: In line94, there will be a judgment to save the checkpoint according to the custom number of steps. The save steps for timeline and timeline both can be customed.

File: modelzoo/template.py (hunk 3 above)
Comment (Duyi-Wang) [reply modelzoo/template.py:38]: Yes, it does create a model. Which should be the output or the ending of the model? Logits(the input of the last sigmod), probability of prediction or prediction?

File: modelzoo/template.py (hunk 1 above)
Comment (Duyi-Wang) [reply modelzoo/template.py:26]: 'input_fn' will be removed in the class. I want to put the forward process into `model_fn`, and put loss computing, optimizer, metrics including ACC & AUC into a function instead of `train_fu` and `metrics_fn`. What to call this function?
TF's estimator API uses a train_op_fn(loss) to create an optimizer, and the loss computing and metrics are packed into a `Head` instance with other things like computing probability of prediction from logits by sigmod().

File: modelzoo/template.py (hunk 5 above)
Comment (Duyi-Wang) [reply modelzoo/template.py]: Feature column is used in model like `net = tf.feature_column.input_layer(features=self.feature, feature_columns=self.deep_column)`, `self.feature` is the input data and the self.deep_column is a list of feature columns describing how to process the corresponding column in input data. Feature column is strongly related to dataset's structure, so I consider it shouldn't be included in model class. In TF's `estimator` API, the feature columns is an input when creating a model.

File: modelzoo/template.py (hunk 6 above)
Comment (Duyi-Wang) [reply modelzoo/template.py:122]: `generate_input_data()` is the `input_fn()`, I will rename it as `input_fn()`. `input_fn()` should return one dataset pipline, we have two dataset, training and test, so it should be called twice. In TF's `estimator` API, `estimator.eval` and `estimator.train` both need an `input_fn()` as input, in the process of training and evaluation, `input_fn()` is used twice as input.

**Comment by Duyi-Wang** [comment]
`Remove train/eval fn` from model class. Changed `loss_computing` to `loss_fn`.

File: modelzoo/template.py [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+        self.logits = None
+        self.probability = tf.math.sigmod(self.logits)
+        self.output = tf.round(self.probability)
+
+    # compute loss
+    def loss_fn(self):
+        self.loss = None
+
+    # define optimizer and generate train_op
+    def train_op_fn(self):
```
Comment (changqi1) [inline modelzoo/template.py:45]: Why is it named train_op_fn?  optimizer_fn ?

File: modelzoo/template.py [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+
+    # create feature column
+    feature_column = build_feature_columns()
+
+    # create variable partitioner for distributed training
+    num_ps_replicas = len(tf_config['ps_hosts']) if tf_config else 0
+    input_layer_partitioner = None
+    dense_layer_partitioner = None
+
+    # create model
```
Comment (changqi1) [inline modelzoo/template.py:123]: # build model
